### PR TITLE
BDD for testing

### PR DESCRIPTION
**Pull request by jcantrill**
This PR is a proof of concept for testing logging using BDD.  The goal is to introduce a framework that

* Better conveys what is being tested then the current tests
* Make CI test progress more obvious
* Introduce a test scripting language that is relatively friendly.

Comments welcome as if this is something we are interested in pursuing.  It will require us to install ruby and associated gems on the devenv.  We possibly could also create an image and drive testing from the image.
This file is probably the one of immediate intrest that corresponds to test/curator.sh https://github.com/openshift/origin-aggregated-logging/pull/574/files#diff-6deaa70b62da401e80de9dcbaee09be2

cc @richm @portante @wozniakjan @josefkarasek @lukas-vlcek @nhosoi @pweil- @mwringe @stevekuznetsov

### CODE REVIEW

**Comment by richm** [comment]
very cool!

**Comment by jcantrill** [comment]
I should also mention this framwork has selenium integration which would facilitate testing of kibana for us.

**Comment by richm** [comment]
This depends on plain old upstream cucumber?  What's missing that is provided by cucushift?  Just the openshift integration/wrappers?

**Comment by jcantrill** [comment]
This has no dependencies on cucushift.  Cucushift could possibly provide:

* Step defs we could reuse
* wrappers around oc

I chose this path based upon my initial evaluation of what we needed for logging and what Cucushift offered in regards to steps; I did not see anything that fit my needs.  It could be I just did not review it in enough depth and warrants re-evaluation

**Comment by richm** [comment]
@jcantrill That's fine - I think this is good because it means we can use cucumber right away - we don't have to wait for cucushift

File: test/cucumber/README.md [hunk 1]
```diff
@@ -10,8 +10,8 @@ $ sudo bash
 $ REMOVE_PACKAGES="ruby-devel wget gcc bzip2 make"  \
   INSTALL_PACKAGES="ruby firefox libffi xorg-x11-server-Xvfb" \
   yum -y $INSTALL_PACKAGES $REMOVE_PACKAGES
-$ gem install bundler --no-ri
-$ bundler install
+$ GEM_HOME=vendor gem install bundler --no-ri
+$ GEM_HOME=vendor bundler install
```
Comment (richm) [inline test/cucumber/README.md]: +1 for using GEM_HOME=vendor

File: test/cucumber.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  os::cmd::expect_success "oc login -u system:admin"
+  os::cmd::expect_success "oadm policy add-cluster-role-to-user cluster-admin admin"
+  os::cmd::expect_success "oc project logging"
+  #create resources
+  os::cmd::expect_success "oc process -f ${CUCUMBER_DIR}/resources.yml | oc apply -f -"
+
+  #build image
+  os::cmd::expect_success "oc start-build logging-cucumber --from-dir=${OS_O_A_L_DIR}"
+  build="logging-cucumber-$(oc get bc logging-cucumber -o jsonpath='{.status.lastVersion}')"
+  os::cmd::try_until_text "oc get build $build -o jsonpath='{ .status.phase }'" "Complete" $CUCUMBER_BUILD_WAIT_TIME
```
Comment (stevekuznetsov) [inline test/cucumber.sh]: Could use

```
oc rollout status buildconfig/name
```

**Pull request closed without merging**